### PR TITLE
Use a http server to hold the port busy before the real test.

### PR DESCRIPTION
**Pull request by rodrigoSaladoAnaya**
Signed-off-by: Rodrigo Salado Anaya <rodrigo.salado.anaya@gmail.com>

Motivation:
Improvement to the test `testListenInvalidPort`, currently when port 80 is available the test fails.

```java
mvn -Dtest=NetTest#testListenInvalidPort test
```

### CODE REVIEW

File: src/test/java/io/vertx/core/net/NetTest.java [hunk 1]
```diff
@@ -903,16 +903,21 @@ public void testConnectInvalidConnectHandler() throws Exception {
...
-    server = vertx.createNetServer(new NetServerOptions().setPort(80));
-    server.connectHandler((netSocket) -> {
-    }).listen(ar -> {
-      assertTrue(ar.failed());
-      assertFalse(ar.succeeded());
-      assertNotNull(ar.cause());
-      testComplete();
+    final int port = 9090;
+    final HttpServer httpServer = vertx.createHttpServer();
+    httpServer.requestHandler(r -> {}).listen(port, httpResult -> {
```
Comment (vietj) [inline src/test/java/io/vertx/core/net/NetTest.java]: can you use the onSuccess / onFailure lambdas to improve the test ?

**Comment by vietj** [comment]
Actually I meant rather the one defined in AsyncTestBase, e.g

```java
server.listen(onSuccess(s ->
   // ...
});
```

**Comment by rodrigoSaladoAnaya** [comment]
Ok, I will review other tests that use them to apply the change.

**Comment by rodrigoSaladoAnaya** [comment]
What do you think @vietj ?

File: src/test/java/io/vertx/core/net/NetTest.java [hunk 2]
```diff
@@ -903,18 +903,18 @@ public void testConnectInvalidConnectHandler() throws Exception {
...
-    }).listen(ar -> {
-      assertTrue(ar.failed());
-      assertFalse(ar.succeeded());
-      assertNotNull(ar.cause());
-      testComplete();
-    });
+    final int port = 9090;
+    final HttpServer httpServer = vertx.createHttpServer()
+      .requestHandler(ignore -> {})
+      .listen(port, httpResult ->
```
Comment (vietj) [inline src/test/java/io/vertx/core/net/NetTest.java]: I think here we need an onSuccess(...) block

**Comment by rodrigoSaladoAnaya** [comment]
I didn't use the method onSuccess, to take advantage of the fact that the port cannot be occupied and the test can still be valid.

Although it is also a good idea to narrow the test as much as possible, ensuring that the port was used before the actual test and released later.

I will apply the suggested change, just the indentation does not convince me.

**Comment by rodrigoSaladoAnaya** [comment]
Hi @vietj, do you agree with the last change?

File: src/test/java/io/vertx/core/net/NetTest.java [hunk 3]
```diff
@@ -903,18 +903,18 @@ public void testConnectInvalidConnectHandler() throws Exception {
...
+      .requestHandler(ignore -> {})
+      .listen(port, onSuccess(s ->
+        vertx.createNetServer()
+          .connectHandler(ignore -> {})
+          .listen(port, onFailure(error -> {
+            assertNotNull(error);
+            testComplete();
+          }))));
     await();
+    httpServer.close();
```
Comment (vietj) [inline src/test/java/io/vertx/core/net/NetTest.java]: we need this close to be in a finally block

**Comment by vietj** [comment]
one more comment @rodrigoSaladoAnaya

**Pull request merged**